src/mlpack/tests/serialization.hpp

```diff
@@ -180,37 +194,43 @@ void SerializeObject(T& t, T& newT)
 {
   std::string fileName = FilterFileName(typeid(T).name());
   std::ofstream ofs(fileName, std::ios::binary);
-  OArchiveType o(ofs);
-
   bool success = true;
-  try
-  {
-    o << BOOST_SERIALIZATION_NVP(t);
-  }
-  catch (boost::archive::archive_exception& e)
+
   {
-    std::cerr << e.what() << std::endl;
-    success = false;
+    OArchiveType o(ofs);
+
+    try
+    {
+      o << BOOST_SERIALIZATION_NVP(t);
+    }
+    catch (boost::archive::archive_exception& e)
+    {
+      std::cerr << e.what() << std::endl;
+      success = false;
+    }
   }
   ofs.close();
 
   BOOST_REQUIRE_EQUAL(success, true);
 
   std::ifstream ifs(fileName, std::ios::binary);
-  IArchiveType i(ifs);
 
-  try
-  {
-    i >> BOOST_SERIALIZATION_NVP(newT);
-  }
-  catch (boost::archive::archive_exception& e)
   {
-    std::cout << e.what() << "\n";
-    success = false;
+    IArchiveType i(ifs);
+
+    try
+    {
+      i >> BOOST_SERIALIZATION_NVP(newT);
+    }
+    catch (boost::archive::archive_exception& e)
+    {
+      std::cout << e.what() << "\n";
+      success = false;
+    }
   }
   ifs.close();
 
-  remove(fileName.c_str());
+//  remove(fileName.c_str());
```

Comment:
Looks like we can remove the line.

Reply:
I actually commented that out for debugging purposes, and should re-add it.  I'll do that and then merge.